packages/crypto/src/public_key.ts

```diff
@@ -1,29 +1,44 @@
-import { Assignable } from '@near-js/types';
 import { baseEncode, baseDecode } from '@near-js/utils';
 import { ed25519 } from '@noble/curves/ed25519';
+import secp256k1 from 'secp256k1';
 
 import { KeySize, KeyType } from './constants';
+import { Assignable } from '@near-js/types';
 
 function key_type_to_str(keyType: KeyType): string {
     switch (keyType) {
         case KeyType.ED25519: return 'ed25519';
+        case KeyType.SECP256K1: return 'secp256k1';
         default: throw new Error(`Unknown key type ${keyType}`);
     }
 }
 
 function str_to_key_type(keyType: string): KeyType {
     switch (keyType.toLowerCase()) {
         case 'ed25519': return KeyType.ED25519;
+        case 'secp256k1': return KeyType.SECP256K1;
         default: throw new Error(`Unknown key type ${keyType}`);
     }
 }
 
+class ED25519PublicKey extends Assignable { keyType: KeyType = KeyType.ED25519; data: Uint8Array; }
+class SECP256K1PublicKey extends Assignable { keyType: KeyType = KeyType.ED25519; data: Uint8Array; }
+
 /**
  * PublicKey representation that has type and bytes of the key.
  */
 export class PublicKey extends Assignable {
-    keyType: KeyType;
-    data: Uint8Array;
+    ed25519Key?: ED25519PublicKey;
+    secp256k1Key?: SECP256K1PublicKey;
```

Comment:
Since only one of the `this.*Key` properties will be set, could we implement getters for `data` and `keyType` on the initialized key pair so that we can continue to expose them?  e.g.
```ts
get keyPair() {
  return this.ed25519Key || this.secp256k1Key;
}

get keyType(): KeyType {
  return this.keyPair.keyType;
}

get data(): Uint8Array {
  return this.keyPair.data;
}

toString(): string {
  const encodedKey = baseEncode(this.data);
  return `${key_type_to_str(this.keyType)}:${encodedKey}`;
}
```

Curious to hear what you think. The borsh-TS mismatch makes it awkward to use these classes now that they're not tightly coupled to ed25519 😅

Reply:
I agree, it will be much better with getters.